Sources/NIOTransportServices/NIOTSEventLoop.swift

```diff
@@ -143,11 +152,12 @@ internal class NIOTSEventLoop: QoSEventLoop {
                 p.fail(EventLoopError.shutdown)
                 return
             }
-            do {
-                p.succeed(try task())
-            } catch {
-                p.fail(error)
-            }
+
+            p.assumeIsolated().completeWith(
+                Result {
+                    try task()
+                }
+            )
```

Comment:
Shouldn't the `scheduleTask` func take a `Sendable` closure? (The other variants do.)

Reply:
Yeah good catch - I think I also missed `@preconcurrency` on all these public methods where I've added `@Sendable` to the closures.

I think I'll still have to do the `assumeIsolated` dance though, because `T` isn't `Sendable`.

Reply:
Yeah, good point.